app/services/foreman/unattended_installation/host_verifier.rb

```diff
@@ -0,0 +1,76 @@
+module Foreman
+  module UnattendedInstallation
+    class HostVerifier
+      attr_reader :errors, :host, :request_ip, :for_host_template, :controller_name
+
+      def initialize(host, request_ip:, for_host_template:)
+        @host = host
+        @errors = []
+        @for_host_template = for_host_template
+        @request_ip = request_ip
+        @controller_name = 'unattended'
+      end
+
+      def valid?
+        return false unless valid_host_token?
```

Comment:
I think its cleaner with an individual return for each case, but I guess thats just personal preference. what do you think ? any reasoning against it ?

Reply:
I'm not a fan of the early return in the positive case. My reasoning is that generally the normal flow is to the end of the function. You only return early on bad situations. In this case it's negative so it can be ok. Just very unused to so many early returns. This may be a "trauma" that some university courses left me. Especially one professor that strongly disliked them because they make formal validation very hard.

tl;dr: mostly personal preference and not used to them utilized in this way so my brain has a hard time parsing it

Reply:
Hmm that is some food for thought :) Actually an OOP purist would cringe at the sight of all these returns. Since in pure OOP a method shall shall only have one single return. 

Got me thinking that this whole class needs re-writing, however for now the point was refactoring the unattended controller itself which i think we have achieved nicely. Maybe for another day/PR ?

Reply:
I lean towards less code changes since this is really a refactoring, it is in the title and it was noted, we want to avoid regressions since this is still fragile code. Let's do this in another round.